### ISSUE TITLE
Fix S2325 FP: Do not raise for methods inside generic class

### ISSUE THREAD

**Issue by pavel-mikula-sonarsource** [SonarSource/sonar-dotnet#3204]
### Description

S2325 is now raised for methods inside generic type.

### Repro steps

``` C#
public class S2325<TFirst, TSecond>
{
    public int BuildSomething()
    {
        return 42;
    }
}
```
`BuildSomething` can be invoked as `instance.BuildSomething`
Changing method into static forces callers to invoke it as `S2325<SomeTypeA, SomeTypeB>.BuildSomething` that makes the code difficult to read.

It's also difficult to use the method from another base class that is not aware of final types yet.

See [CA1000](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1000?view=vs-2019)
See [this case](https://sonarcloud.io/project/issues?id=sonaranalyzer-dotnet&issues=AXCtNZOzAFhjSEJCuYhA&open=AXCtNZOzAFhjSEJCuYhA)

### Related information

* SonarC# Version 8.4
